### ISSUE TITLE
No handlers could be found for logger "root"

### ISSUE THREAD

**Issue by jonathanstegall** [anthonylv/pyD2W#4]
I've installed pyD2W in a virtualenvironment (running python 2.7.11 on OSX 10.11.4 with Homebrew). When I try to run any of the d2w.py commands, I get the following error:

```
No handlers could be found for logger "root"
Traceback (most recent call last):
  File "d2w.py", line 442, in <module>
    main(sys.argv[1:])
  File "d2w.py", line 402, in main
    setup_logging(settings['d2w'])
KeyError: 'd2w'
```

I tried commenting out all the call to `setup_logging` (since I'm running it locally anyway) and I got the following:

```
No handlers could be found for logger "root"
The migration process will alter your database
Are you sure you want to continue? [y/N] y
Traceback (most recent call last):
  File "d2w.py", line 442, in <module>
    main(sys.argv[1:])
  File "d2w.py", line 433, in main
    process_action(settings, action, options)
  File "d2w.py", line 383, in process_action
    process_migration(settings, selected_database)
  File "d2w.py", line 289, in process_migration
    database = settings['database']['drupal_database']
KeyError: 'database'
```

I do have my settings configured correctly, I think:

```
database:
    drupal_host: localhost
    drupal_username: root
    drupal_password: 
    drupal_database: drupal
    wordpress_host: localhost
    wordpress_username: root
    wordpress_password: 
    wordpress_database: wordpress
    # A user with CREATE privilege to create the tables
    admin_username: (wordpressadmin)
    admin_password: (wordpressadminpassword)
```

**Comment by anthonylv** [comment]
Hi,

It looks like the options to set up logging are missing from the settings file. Do you have the following in your settings.yml?

```
d2w:
    log_filename: log.txt
    log_max_bytes: 1048576
    log_backup_count: 5
```

**Comment by jonathanstegall** [comment]
@anthonylv Thanks for replying! Yes, those settings are unchanged. Do I need to set a filepath for log.txt or anything?

**Comment by anthonylv** [comment]
@jonathanstegall no worries. The error message indicates that it can't find the `d2w` and `database` option groups in the settings file. Did the script find the actual settings file? By default it will look for the file `settings.yml`. Try either:
1. Making sure the `settings.yml` file exists in the script's directory.
2. Hardcoding the full path to your settings file in `def get_settings()`

I've just noticed a silly bug where I try to write errors to the log before I've set up logging. It's possible the script couldn't find the file for some reason and doesn't display that error. I'm using OS X but not homebrew so maybe there are differences in the way it searches for the file paths.
